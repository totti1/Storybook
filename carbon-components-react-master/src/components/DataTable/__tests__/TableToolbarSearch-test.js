import React from 'react';
import { mount } from 'enzyme';
import { TableToolbarSearch } from '../';

describe('DataTable.TableToolbarSearch', () => {
  it('should render', () => {
    const wrapper = mount(
      <TableToolbarSearch className="custom-class" onChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
