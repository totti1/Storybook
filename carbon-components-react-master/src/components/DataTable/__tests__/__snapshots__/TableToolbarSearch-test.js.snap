// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableToolbarSearch should render 1`] = `
<TableToolbarSearch
  className="custom-class"
  onChange={[MockFunction]}
>
  <div
    className="bx--toolbar-search-container"
  >
    <Search
      className="custom-class"
      id="search-2"
      labelText="Filter table"
      onChange={[MockFunction]}
      placeHolderText="Search"
      small={true}
      type="text"
    >
      <div
        className="bx--search bx--search-with-options bx--search--sm custom-class"
        role="search"
      >
        <Icon
          className="bx--search-magnifier"
          description="search"
          fillRule="evenodd"
          name="search--glyph"
          role="img"
        >
          <svg
            alt="search"
            aria-label="search"
            className="bx--search-magnifier"
            fillRule="evenodd"
            height="16"
            name="search--glyph"
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <title>
              search
            </title>
            <path
              d="M6 2c2.2 0 4 1.8 4 4s-1.8 4-4 4-4-1.8-4-4 1.8-4 4-4zm0-2C2.7 0 0 2.7 0 6s2.7 6 6 6 6-2.7 6-6-2.7-6-6-6zm10 13.8L13.8 16l-3.6-3.6 2.2-2.2z"
              key="key0"
            />
            <path
              d="M16 13.8L13.8 16l-3.6-3.6 2.2-2.2z"
              key="key1"
            />
          </svg>
        </Icon>
        <label
          className="bx--label"
          htmlFor="search-2"
        >
          Filter table
        </label>
        <input
          className="bx--search-input"
          id="search-2"
          onChange={[Function]}
          placeholder="Search"
          type="text"
        />
        <Icon
          className="bx--search-close bx--search-close--hidden"
          description="close"
          fillRule="evenodd"
          name="close--glyph"
          onClick={[Function]}
          role="img"
        >
          <svg
            alt="close"
            aria-label="close"
            className="bx--search-close bx--search-close--hidden"
            fillRule="evenodd"
            height="16"
            name="close--glyph"
            onClick={[Function]}
            role="img"
            viewBox="0 0 16 16"
            width="16"
          >
            <title>
              close
            </title>
            <path
              d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm3.5 10.1l-1.4 1.4L8 9.4l-2.1 2.1-1.4-1.4L6.6 8 4.5 5.9l1.4-1.4L8 6.6l2.1-2.1 1.4 1.4L9.4 8l2.1 2.1z"
              key="key0"
            />
          </svg>
        </Icon>
      </div>
    </Search>
  </div>
</TableToolbarSearch>
`;
